src/easylink/configuration.py

```diff
@@ -162,72 +155,27 @@ def spark_resources(self) -> Dict[str, Any]:
-        return self.pipeline[step_name].implementation.name
-
     #################
     # Setup Methods #
     #################
-    @staticmethod
-    def _spark_is_required(pipeline: Dict[str, Any]) -> bool:
-        """Check if the pipeline requires spark resources."""
-        implementation_metadata = load_yaml(paths.IMPLEMENTATION_METADATA)
-        try:
-            implementations = [step["implementation"]["name"] for step in pipeline.values()]
-        except Exception as e:
-            raise KeyError(
-                "The pipeline specification should contain keys for each step "
-                "and each step should contain an 'implementation' key with a 'name' key."
-            ) from e
-        for implementation in implementations:
-            if implementation_metadata[implementation].get("requires_spark", False):
-                return True
-        return False
+    def update_implementation_configs(self, level: LayeredConfigTree):
+        if "implementation" in level:
+            level.implementation.update({"configuration": {}}, layer="default")
+        else:
+            for sub_level in level.values():
+                if isinstance(sub_level, LayeredConfigTree):
+                    self.update_implementation_configs(sub_level)
 
     def _get_schema(self) -> Optional[PipelineSchema]:
         """Validates the pipeline against supported schemas.
 
-        NOTE: this acts as the pipeline configurat file's validation method since
+        NOTE: this acts as the pipeline configuration file's validation method since
         we can only find a matching schema if the file is valid.
         """
         errors = defaultdict(dict)
-
-        # Check that each of the pipeline steps also contains an implementation
-        metadata = load_yaml(paths.IMPLEMENTATION_METADATA)
-        for step, step_config in self.pipeline.items():
-            if not "name" in step_config["implementation"]:
-                errors[PIPELINE_ERRORS_KEY][
-                    f"step {step}"
-                ] = "The implementation does not contain a 'name'."
-            elif not step_config["implementation"]["name"] in metadata:
-                errors[PIPELINE_ERRORS_KEY][f"step {step}"] = (
-                    f"Implementation '{step_config['implementation']['name']}' is not supported. "
-                    f"Supported implementations are: {list(metadata.keys())}."
-                )
-        if errors:
-            exit_with_validation_error(dict(errors))
-
         # Try each schema until one is validated
         for schema in PIPELINE_SCHEMAS:
-            logs = []
-            config_steps = self.pipeline.keys()
-            # Check that number of schema steps matches number of implementations
-            if len(schema.steps) != len(config_steps):
-                logs.append(
-                    f"Expected {len(schema.steps)} steps but found {len(config_steps)} implementations. "
```

Comment:
we validate that we have all the needed steps and no extraneous ones in `step.py` now.